### PR TITLE
[#9084] Data Migration: FeedbackSession timeZone, all time fields, is*EmailEnabled and feedbackSessionType

### LINKED ISSUE #9084: Data Migration: FeedbackSession timeZone, all time fields, is*EmailEnabled and feedbackSessionType
Data migration initiated by issue #8448

- [x] Merged code that work for both the old and new versions of the data schema.
- [x] Released a version containing the above in [V6.5.0](https://github.com/TEAMMATES/teammates/releases/tag/V6.5.0).
- [x] Created a client script to migrate all data following the old schema to the new schema in PR #9081 and #8759.
- [x] Ran the script against the live site.
- [ ] Removed the code that is specifically tailored for the old data schema in PR #.

### PR DESCRIPTION
**Pull request by xpdavid**
Fixes #9084 

Pre-requisite for #9088 

Removed all hack code for legacy data in `FeedbackSession`. 🎉 

I don't think it is necessary to write another script just to delete unused fields (< 1MB extra space) in the LIVE database. Any future data migration to `FeedbackSession` will do the job automatically. 

Note to reviewer:

- Checked that all test files in the repo follow the new format without new fields missing and old fields appearing.

### CODE REVIEW

File: src/test/resources/data/InstructorFeedbackEditPageUiTest.json [hunk 1]
```diff
@@ -321,7 +321,6 @@
       "resultsVisibleFromTime": "2026-05-01T21:59:00Z",
       "timeZone": "Australia/Victoria",
       "gracePeriod": 10,
-      "feedbackSessionType": "STANDARD",
```
Comment (wkurniawan07) [inline src/test/resources/data/InstructorFeedbackEditPageUiTest.json]: What is this doing?

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java [hunk 2]
```diff
@@ -34,47 +32,24 @@
...
-            boolean isStartTimeWithinRange = (fsStart.isAfter(rangeStart) || fsStart.equals(rangeStart))
-                    && fsStart.isBefore(rangeEnd);
-            boolean isEndTimeWithinRange = fsEnd.isAfter(rangeStart) && (fsEnd.isBefore(rangeEnd) || fsEnd.equals(rangeEnd));
-
-            if (isStartTimeWithinRange || isEndTimeWithinRange) {
-                list.add(fs);
-            }
-        }
+        endEntities.removeAll(startEntities);
+        endEntities.addAll(startEntities);
```
Comment (wkurniawan07) [inline src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: Is this trying to de-duplicate? If so you may want to consider using `Set` instead, although this works.

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java [hunk 3]
```diff
@@ -34,47 +32,24 @@
...
+    /**
+     * Gets a list of feedback sessions that have start/end time laid between {@code rangeStart} and {@code rangeEnd}.
+     */
     public List<FeedbackSessionAttributes> getAllOpenFeedbackSessions(Instant rangeStart, Instant rangeEnd) {
-        List<FeedbackSessionAttributes> list = new LinkedList<>();
-
-        // To retrieve legacy data where local dates are stored instead of UTC
-        // TODO: remove after all legacy data has been converted
-        Instant start = rangeStart.minus(Duration.ofHours(25));
-        Instant end = rangeEnd.plus(Duration.ofHours(25));
```
Comment (wkurniawan07) [inline src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: This was to account for possible timezone offsets before. Although I trust the timezone-handling code, can confirm that it's safe to remove these?

File: src/test/resources/data/InstructorFeedbackEditPageUiTest.json (hunk 1 above)
Comment (xpdavid) [reply src/test/resources/data/InstructorFeedbackEditPageUiTest.json]: `feedbackSessionType` attribute is removed as data migration completed. There is no point to keep the attribute in test JSON data.

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java (hunk 3 above)
Comment (xpdavid) [reply src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: After running script in #9084, all entities should be stored in UTC right now.

File: src/main/java/teammates/storage/api/FeedbackSessionsDb.java (hunk 2 above)
Comment (xpdavid) [reply src/main/java/teammates/storage/api/FeedbackSessionsDb.java]: A bit overkill to wrap the returned list to set and set to list again just to remove duplications. Will add comment here instead.

**Comment by xpdavid** [comment]
@tran-tien-dat wait for your comment :) Will request final review tmr if there is no objection.

**Pull request merged**